sources/packages/backend/workflow/test/2022-2023/parttime-assessment/costs/parttime-assessment-2022-2023-costs-child-care-cost.e2e-spec.ts

```diff
@@ -0,0 +1,211 @@
+    },
+  );
+
+  it(
+    "Should calculate child care costs as maximum allowed limit when student has one or more dependents on both categories " +
+      "(11 years or under and over 12 years or over and declared on taxes for disability) and " +
+      "child care costs entered is beyond maximum allowable limit for the " +
+      "given number of dependents, course load and offering weeks.",
+    async () => {
+      // Arrange
+      const assessmentConsolidatedData =
+        createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+      assessmentConsolidatedData.studentDataDaycareCosts11YearsOrUnder = 1000;
+      assessmentConsolidatedData.studentDataDaycareCosts12YearsOrOver = 1000;
+      assessmentConsolidatedData.offeringWeeks = 17;
+      assessmentConsolidatedData.offeringCourseLoad = 20;
+      // Creates 2 eligible dependents.
+      assessmentConsolidatedData.studentDataDependants = [
+        createFakeStudentDependentEligibleForChildcareCost(
+          DependentChildCareEligibility.Eligible0To11YearsOld,
+          assessmentConsolidatedData.offeringStudyStartDate,
+        ),
+        createFakeStudentDependentEligibleForChildcareCost(
+          DependentChildCareEligibility.Eligible12YearsAndOver,
+          assessmentConsolidatedData.offeringStudyStartDate,
+        ),
+      ];
+      const totalChildcareCostEnteredInApplication =
+        assessmentConsolidatedData.studentDataDaycareCosts11YearsOrUnder +
+        assessmentConsolidatedData.studentDataDaycareCosts12YearsOrOver;
+      // Act
+      const calculatedAssessment =
+        await executePartTimeAssessmentForProgramYear(
+          PROGRAM_YEAR,
+          assessmentConsolidatedData,
+        );
+      // Assert
+      // Childcare cost for the category 11 years and under must be equal to the value entered in application
+      // for this category as student has one or more dependent in this category.
+      expect(
+        calculatedAssessment.variables.calculatedDataDaycareCosts11YearsOrUnder,
+      ).toBe(assessmentConsolidatedData.studentDataDaycareCosts11YearsOrUnder);
+      // Childcare cost for the category 12 years and over under must be equal to the value entered in application
+      // for this category as student has one or more dependent in this category.
+      expect(
+        calculatedAssessment.variables.calculatedDataDaycareCosts12YearsOrOver,
+      ).toBe(assessmentConsolidatedData.studentDataDaycareCosts12YearsOrOver);
+      // Child care cost expected to be the total sum of values entered in student application.
+      expect(calculatedAssessment.variables.calculatedDataChildCareCost).toBe(
+        totalChildcareCostEnteredInApplication,
+      );
+      // As the total sum of child care costs are beyond the maximum allowable limit
+      // the maximum limit must be considered towards assessed need.
+      expect(
+        calculatedAssessment.variables.calculatedDataTotalChildCareCost,
+      ).toBe(1822.4);
+      expect(
```

Comment:
Isn't this expect redundant? If not, then we can add a similar expect in the first test above.

Reply:
The second expect is being explicit about **considered child care costs** being less than total childcare costs entered in student application application(when exceeding maximum limit) .

The first test does not have child care cost exceeding maximum limit and hence it is not applicable to it.

Reply:
Thanks for the clarification @dheepak-aot